### PR TITLE
Fix/checkbox typescript

### PR DESCRIPTION
**Pull request by jwinding**
## Explain the changes you’ve made

Moved some size and style settings into the theme, and converted the Checkbox atom to typescript.
Also remove a bunch of old colors for the checkbox that was no longer used. 

## Explain why these changes are made

For consistency, we want all our styling parameters in the theme. And typescript is better than no typescript. 

## How to test the changes?

Check any story or form which has a checkbox. Nothing should be changed in how it works 

## Was this feature tested in the following environments?
- [x] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

### CODE REVIEW

File: source/components/atoms/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -19,48 +29,24 @@ const CheckboxTick = styled(Icon)`
...
   const theme = useContext(ThemeContext);
-  const style = {
-    ...sizes[size],
-  };
-  const validColorSchema = getValidColorSchema(color);
-  const tickIcon = <CheckboxTick size={iconSizes[size]} name="done" />;
+
+  // some type trickery to make typescript happy
+  let validColorSchema: 'blue' | 'green' | 'red' | 'purple'; 
+  const vCS = getValidColorSchema(color);
```
Comment (dannil76) [inline source/components/atoms/Checkbox/Checkbox.tsx:45]: meehh... what is vCS?

Comment (dannil76) [reply source/components/atoms/Checkbox/Checkbox.tsx:45]: Works well otherwise

Comment (dannil76) [reply source/components/atoms/Checkbox/Checkbox.tsx:45]: One more thing... Emojis in commit messages is REQUIRED

File: source/components/atoms/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -19,48 +29,24 @@ const CheckboxTick = styled(Icon)`
...
+const Checkbox: React.FC<Props> = ({checked, onChange, size, disabled, color}) => {
   const theme = useContext(ThemeContext);
-  const style = {
-    ...sizes[size],
-  };
-  const validColorSchema = getValidColorSchema(color);
-  const tickIcon = <CheckboxTick size={iconSizes[size]} name="done" />;
+
+  // some type trickery to make typescript happy
+  let validColorSchema: 'blue' | 'green' | 'red' | 'purple';
```
Comment (TeddyAndersson) [inline source/components/atoms/Checkbox/Checkbox.tsx:44]: This could be moved to a stand alone type since it's used in several components at this stage :)

**Review by TeddyAndersson — CHANGES_REQUESTED** [review]

File: source/components/atoms/Checkbox/Checkbox.tsx (hunk 2 above)
Comment (TeddyAndersson) [reply source/components/atoms/Checkbox/Checkbox.tsx:44]: Even in several places within this component :)

Comment (jwinding) [reply source/components/atoms/Checkbox/Checkbox.tsx:44]: Yeah, agreed, I was thinking the same thing but not sure where to put it. I think I'll make a separate typescript file in the styles folder that can export this type, as well as a typed version of the getValidColorSchema function.

Comment (TeddyAndersson) [reply source/components/atoms/Checkbox/Checkbox.tsx]: Yeah that feels like a good idea (y)

Since that's the origin of the type :D

Comment (jwinding) [reply source/components/atoms/Checkbox/Checkbox.tsx]: I have now done exactly that, added a themeHelpers.ts which contains two color types and a typed getValidColorSchema function.

**Review by TeddyAndersson — APPROVED** [review]

**Pull request merged**